src/lib/tachyon-utils.js

```diff
@@ -273,11 +278,103 @@ async function getIdentification({ deviceId, partitionTable, partitionFilenames
 	};
 }
 
+async function promptWifiNetworks(ui = new UI()) {
+	let pickedSSID;
+	ui.write(
+		ui.chalk.bold(
+			`Wi-Fi setup is required to continue.${os.EOL}` +
+			`An active internet connection is necessary to activate cellular connectivity on your device.${os.EOL}`
+		)
+	);
+
+	const networks = await ui.showBusySpinnerUntilResolved(
+		'Scanning for nearby Wi-Fi networks...',
+		_scanNetworks()
+	);
+	const ssids = networks.map(n => n.ssid);
+	const otherNetworkLabel = '[Other Network]';
+	const question = [
+		{
+			type: 'list',
+			name: 'ssid',
+			message: chalk.bold.white('Select the Wi-Fi network with which you wish to connect your device:'),
+			choices: () => {
+				const ns = ssids.slice();
+				ns.unshift(new inquirer.Separator());
+				ns.unshift('[Other Network]');
+				ns.unshift(new inquirer.Separator());
+				return ns;
+			},
+			when: () => {
+				return networks.length;
```

Comment:
If no SSIDs are detected, this will skip the question but won't ask for the SSID, right?

Reply:
ohh right! if is null should ask for manual entry

Reply:
Fixed here! https://github.com/particle-iot/particle-cli/pull/834/commits/60c7df7b37a60433ca539f2087da2e9aee504fc4